lib/declare_schema/model/index_definition.rb

```diff
@@ -69,12 +71,30 @@ def default_index_name(table, fields)
 
         private
 
+        SHA_SUFFIX_LENGTH = 4
+
+        def shorten_name(name, max_len)
+          if name.size <= max_len
+            name
+          else
+            name_prefix = name.first(max_len >= SHA_SUFFIX_LENGTH*2 ? (max_len - SHA_SUFFIX_LENGTH) : ((max_len + 1)/2))
+            sha = Digest::SHA256.hexdigest(name)
+            (name_prefix + sha).first(max_len)
+          end
+        end
+
         def long_index_name(table_name, columns)
           "index_#{table_name}_on_#{Array(columns).join("_and_")}"
         end
 
         def short_index_name(table_name, columns)
-          "#{table_name}__#{Array(columns).join("_")}"
+          columns_suffix = "__" + Array(columns).join('_')
+          if DeclareSchema.max_index_and_constraint_name_length.nil?
+            table_name + columns_suffix
+          else
+            max_name_len = [DeclareSchema.max_index_and_constraint_name_length - columns_suffix.length, 0].max
+            shorten_name(table_name, max_name_len) + columns_suffix
```

Comment:
If this is still too long, are we ok with the expectation that users need to declare a name?

Reply:
Yes, I think that's the best we can do.

Reply:
...especially now that the error message suggests how to fix it. :)